<a href="https://colab.research.google.com/github/OpheliaVlzqz/MineriaDatos/blob/main/Ej_Limpieza_Equipo10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Equipo: 10
###Integrantes


* 1856652 Velázquez Montemayor Ofelia

* 1862149 Cruz Martínez Diego Adrian

* 1867216 Martínez Reyes Jose Miguel

* 1941488 García Ortega Jose Andrés 

* 1941564 Ramirez Moreno Diana Gabriela 



###Grupo 001 Lunes

Nombre de la base de datos: Google Play Store Apps


URL de la pagina: https://www.kaggle.com/lava18/google-play-store-apps



Objetivo: Facilitar al usuario a encontrar aplicaciones idóneas con sugerencias basadas en sus búsquedas recientes o su ineteraccion en general con la plataforma. 

##LIMPIEZA DE DATOS

**Leer el archivo**

EL primer paso sera importar las librerias las cuales podremos utilizar para la lectura y limpieza de datos. Despues se mandará llamar el archivo el cual esta en formato excel. Para eliminar los posibles simbolos que no se puedan leer utilizaremos encoding.


In [ ]:
import pandas as pd
import numpy as np

google = pd.read_csv("/content/googleplaystore.csv")

**Tamaño de la base de datos**

para revidar el tamaño de la base utilizaremos la funcion shape.

In [ ]:
google.shape

(10841, 13)

Podemos ver que nuestra base de datos cuenta con 10,841 filas (aplicaciones) y 13 columnas.

Para revisar nuestros primeros 10 datos de la base de datos se utilizara la funcion head y para los ultimos sera la funcion tail

In [ ]:
google.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

In [ ]:
google.tail(10)

App  ...         Android Ver
10831                       payermonstationnement.fr  ...          4.0 and up
10832                                       FR Tides  ...          2.1 and up
10833                                    Chemin (fr)  ...          2.2 and up
10834                                  FR Calculator  ...          4.1 and up
10835                                       FR Forms  ...          4.0 and up
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

**Manejo de Columnas**

Para facilitar nuestra visualización de los datos nosotros decidiremos eliminar columnas las cuales no nos den una informacion relevante para nuestro objetivo. para decidir que columnas eliminaremos primero veremos el nombre de las columnas que tenemos y despues el tipo de dato que contienen


In [ ]:
google.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [ ]:
google.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

Podemos ver que existen diferentes columnas las cuales no nos serviran para nuestro objetivo y el tenerlas mostradas solo entorpecera la visualizacion de nuestros datos, por lo cual las eliminaremos.


In [ ]:
google1=google.drop(['Size','Price','Genres','Last Updated','Current Ver','Android Ver'],axis=1)
google1

App  ... Content Rating
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...       Everyone
1                                    Coloring book moana  ...       Everyone
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...       Everyone
3                                  Sketch - Draw & Paint  ...           Teen
4                  Pixel Draw - Number Art Coloring Book  ...       Everyone
...                                                  ...  ...            ...
10836                                   Sya9a Maroc - FR  ...       Everyone
10837                   Fr. Mike Schmitz Audio Teachings  ...       Everyone
10838                             Parkinson Exercices FR  ...       Everyone
10839                      The SCP Foundation DB fr nn5n  ...     Mature 17+
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...       Everyone

[10841 rows x 7 columns]

Ahora cambiaremos el nombre de las columnas para que nos sea mas facil identificar de que esta hablando cada columna.


In [ ]:
google1.rename(columns={'App':'Nombre de la Aplicacion'}, inplace = True)
google1.rename(columns={'Category':'Categoria'},inplace = True)
google1.rename(columns={'Rating':'Calificacion'},inplace = True)
google1.rename(columns={'Reviews':'Numero de reseñas'},inplace = True)
google1.rename(columns={'Installs':'Instaladas'},inplace = True)
google1.rename(columns={'Type':'Gratis/Pago'},inplace = True)
google1.rename(columns={'Content Rating':'Clasificacion'},inplace = True)

google1.head()

Nombre de la Aplicacion  ... Clasificacion
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...      Everyone
1                                Coloring book moana  ...      Everyone
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...      Everyone
3                              Sketch - Draw & Paint  ...          Teen
4              Pixel Draw - Number Art Coloring Book  ...      Everyone

[5 rows x 7 columns]

Revisaremos si existen columnas con datos nulos.

In [ ]:
google1.isnull().any()

Nombre de la Aplicacion    False
Categoria                  False
Calificacion                True
Numero de reseñas          False
Instaladas                 False
Gratis/Pago                 True
Clasificacion               True
dtype: bool

In [ ]:
google1.isnull().sum()

Nombre de la Aplicacion       0
Categoria                     0
Calificacion               1474
Numero de reseñas             0
Instaladas                    0
Gratis/Pago                   1
Clasificacion                 1
dtype: int64

Dado que si existen valores nulos en nuestras filas y estos datos nos apoyaran a llegar a nuestro obetivo los eliminaremos.

In [ ]:
g2=google1.dropna()

Revisamos si funcionó

In [ ]:
g2.isnull().any()

Nombre de la Aplicacion    False
Categoria                  False
Calificacion               False
Numero de reseñas          False
Instaladas                 False
Gratis/Pago                False
Clasificacion              False
dtype: bool

Ordenaremos las columnas de forma que los datos mas relevantes, para nosotros, se presenten primero.

In [ ]:
#https://www.delftstack.com/es/howto/python-pandas/how-to-change-the-order-of-dataframe-columns/
#https://www.delftstack.com/es/howto/python-pandas/pandas-remove-index/#:~:text=Si%20queremos%20eliminar%20la%20columna,en%20el%20m%C3%A9todo%20reset_index()%20.
g3=g2.reindex(columns=['Nombre de la Aplicacion','Calificacion','Numero de reseñas','Instaladas','Categoria','Clasificacion','Gratis/Pago'])
g3=g3.set_index('Nombre de la Aplicacion')
g3

Calificacion  ... Gratis/Pago
Nombre de la Aplicacion                                           ...            
Photo Editor & Candy Camera & Grid & ScrapBook               4.1  ...        Free
Coloring book moana                                          3.9  ...        Free
U Launcher Lite – FREE Live Cool Themes, Hide Apps           4.7  ...        Free
Sketch - Draw & Paint                                        4.5  ...        Free
Pixel Draw - Number Art Coloring Book                        4.3  ...        Free
...                                                          ...  ...         ...
FR Calculator                                                4.0  ...        Free
Sya9a Maroc - FR                                             4.5  ...        Free
Fr. Mike Schmitz Audio Teachings                             5.0  ...        Free
The SCP Foundation DB fr nn5n                                4.5  ...        Free
iHoroscope - 2018 Daily Horoscope & Astrology                4.5  ...        Free

[9366 rows x 6 columns]

Ahora tomaremos solo las calificaciones que tienen una buena calificacion (mayor a 3).

In [ ]:
g4=g3[g3['Calificacion'] > 3]
g4

Calificacion  ... Gratis/Pago
Nombre de la Aplicacion                                           ...            
Photo Editor & Candy Camera & Grid & ScrapBook               4.1  ...        Free
Coloring book moana                                          3.9  ...        Free
U Launcher Lite – FREE Live Cool Themes, Hide Apps           4.7  ...        Free
Sketch - Draw & Paint                                        4.5  ...        Free
Pixel Draw - Number Art Coloring Book                        4.3  ...        Free
...                                                          ...  ...         ...
FR Calculator                                                4.0  ...        Free
Sya9a Maroc - FR                                             4.5  ...        Free
Fr. Mike Schmitz Audio Teachings                             5.0  ...        Free
The SCP Foundation DB fr nn5n                                4.5  ...        Free
iHoroscope - 2018 Daily Horoscope & Astrology                4.5  ...        Free

[8996 rows x 6 columns]

Por ultimo ordenaremos los datos de la calificacion de manera descendente para asi tener las aplicaciones con mejor puntuacion primero.

In [ ]:
#https://www.analyticslane.com/2019/04/29/diferentes-formas-de-ordenar-dataframes-en-pandas/

google_final=g4.sort_values('Calificacion',ascending=False)

In [ ]:
google_final.head()

Calificacion  ... Gratis/Pago
Nombre de la Aplicacion                       ...            
Quran Khmer Offline AY                   5.0  ...        Free
EK Bailey Preaching Conference           5.0  ...        Free
Spine- The dating app                    5.0  ...        Free
Awake Dating                             5.0  ...        Free
American Girls Mobile Numbers            5.0  ...        Free

[5 rows x 6 columns]

In [ ]:
google_final.tail()

Calificacion  ... Gratis/Pago
Nombre de la Aplicacion                    ...            
Clear                                 3.1  ...        Free
My Telcel                             3.1  ...        Free
EU VAT Checker                        3.1  ...        Free
EF English Live for tablets           3.1  ...        Free
EF Teacher                            3.1  ...        Free

[5 rows x 6 columns]

Guardaremos nuestro nuevo data frame en un archivo nuevo.

In [ ]:
#https://stackoverflow.com/questions/53898836/export-dataframe-as-csv-file-from-google-colab-to-google-drive/60733479

from google.colab import drive
drive.mount('drive')

Mounted at drive


In [ ]:
google_final.to_csv('Google_Final.csv')
!cp Google_Final.csv "drive/My Drive/"

El documento ha quedado guardado

##Preguntas de interes

**¿Para que me puede servir estas agrupaciones para la practica siguiente?**

Para poder delimitar mejor nuestra informacion. Se podra tener solo los datos que buscamos que cumplan con cierto parametro.


**Importancia del analisis previo de los datos antes de iniciar las modificaciones.**

Conocer nuestros datos preeviamente nos facilitara el proceso ya que tendremos bien delimitado lo que necesitamos, a donde queremos llegar y con eso hacer lo necesario para tener lo esencial de la manera mas optima.

##Puntos Extra

**Investigar los siguientes comandos: ?df.dropna() ?df.astype**

* df.dropna() https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html
    
> > Elimina toda la fila o coumna en la cual no se presente algun dato("NaN").

* df.astype() 

> > Esta funcion cambia el tipo de datos de toda la tabla o en su defecto solo las columnas seleccionadas.



In [ ]:
google_final.dropna()

Calificacion  ... Gratis/Pago
Nombre de la Aplicacion                       ...            
Quran Khmer Offline AY                   5.0  ...        Free
EK Bailey Preaching Conference           5.0  ...        Free
Spine- The dating app                    5.0  ...        Free
Awake Dating                             5.0  ...        Free
American Girls Mobile Numbers            5.0  ...        Free
...                                      ...  ...         ...
Clear                                    3.1  ...        Free
My Telcel                                3.1  ...        Free
EU VAT Checker                           3.1  ...        Free
EF English Live for tablets              3.1  ...        Free
EF Teacher                               3.1  ...        Free

[8996 rows x 6 columns]

In [ ]:
g5= g4.astype("object")
g5.dtypes

Calificacion         object
Numero de reseñas    object
Instaladas           object
Categoria            object
Clasificacion        object
Gratis/Pago          object
dtype: object

**Realizar una tarea de preprocesamiento no mencionada en esta práctica.**

Agruparemos las categorias y encontraremos el promedio de calificacion de cada una de ellas

In [ ]:
#https://datacarpentry.org/python-ecology-lesson-es/02-starting-with-data/
genres_count=google_final.groupby('Categoria').mean()
genres_count

Calificacion
Categoria                        
ART_AND_DESIGN           4.358065
AUTO_AND_VEHICLES        4.260000
BEAUTY                   4.278571
BOOKS_AND_REFERENCE      4.355367
BUSINESS                 4.248571
COMICS                   4.200000
COMMUNICATION            4.201875
DATING                   4.123864
EDUCATION                4.389032
ENTERTAINMENT            4.133784
EVENTS                   4.470455
FAMILY                   4.253595
FINANCE                  4.269333
FOOD_AND_DRINK           4.224528
GAME                     4.302574
HEALTH_AND_FITNESS       4.415942
HOUSE_AND_HOME           4.216000
LIBRARIES_AND_DEMO       4.178462
LIFESTYLE                4.255789
MAPS_AND_NAVIGATION      4.142241
MEDICAL                  4.329321
NEWS_AND_MAGAZINES       4.217195
PARENTING                4.346939
PERSONALIZATION          4.346154
PHOTOGRAPHY              4.260726
PRODUCTIVITY             4.279290
SHOPPING                 4.290171
SOCIAL                   4.282353
SPORTS                   4.260256
TOOLS                    4.173824
TRAVEL_AND_LOCAL         4.177315
VIDEO_PLAYERS            4.137908
WEATHER                  4.244000